packages/lib/delivery/src/Session.ts

```diff
@@ -57,48 +53,28 @@ export async function checkToken(
         const jwtPayload = verify(token, serverSecret, {
             algorithms: ['HS256'],
         });
-        console.log('jwt string', jwtPayload);
-        console.log(decode(token));
-
-        // check if type is string -> invalid
-        if (typeof jwtPayload == 'string') {
-            logDebug({
-                text: `jwt invalid: jwtPayload is a string`,
-            });
-            return false;
-        }
+        console.log('jwt payload', decode(token));
 
-        // check if expected fields are present
+        // check if payload is well formed
         if (
-            !('account' in jwtPayload) ||
-            !('iat' in jwtPayload) ||
-            !('exp' in jwtPayload) ||
-            !('nbf' in jwtPayload)
+            typeof jwtPayload == 'string' ||
```

Comment:
Checking for strict equality should always preferred to avoid unintended behavior.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Reply:
right, fixed